src/agent/onefuzz/src/fs.rs

```diff
@@ -170,6 +172,104 @@ impl OwnedDir {
     }
 }
 
+#[cfg(target_os = "linux")]
+pub async fn _sync(
+    src: impl AsRef<OsStr>,
+    dst: impl AsRef<OsStr>,
+    delete_dst: bool,
+    recursive: bool,
+) -> Result<()> {
+    let mut cmd = Command::new("rsync");
+    cmd.kill_on_drop(true)
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .arg(if recursive { "-zhr" } else { "-zh" });
+
+    if delete_dst {
+        cmd.arg("--delete");
+    }
+    cmd.arg(&src).arg(&dst);
+
+    let output = cmd
+        .spawn()
+        .context("rsync failed to start")?
+        .wait_with_output()
+        .await
+        .context("rsync failed to run")?;
+
+    if !output.status.success() {
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        anyhow::bail!(
+            "sync failed src:{:?} dst:{:?} stdout:{:?} stderr:{:?}",
+            src.as_ref(),
+            dst.as_ref(),
+            stdout,
+            stderr
+        );
+    }
+
+    Ok(())
+}
+
+#[cfg(target_os = "windows")]
+pub async fn _sync(
+    src: impl AsRef<OsStr>,
+    dst: impl AsRef<OsStr>,
+    delete_dst: bool,
+    recursive: bool,
+) -> Result<()> {
+    let mut cmd = Command::new("robocopy");
```

Comment:
Probably not for this PR, but we probably want to do early-checking the platform specific tools (`robocoby` or `rsync`) from here, and `cdb` or `gdb` for the crash reporting tasks during initialization.

Reply:
creates #669 to track this